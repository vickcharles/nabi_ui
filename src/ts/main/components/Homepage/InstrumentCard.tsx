import * as React from 'react';
import  Card  from 'material-ui/Card';
import { CardContent } from 'material-ui/Card';
import CardMedia from 'material-ui/Card';
import Typography from 'material-ui/Typography';
import Button from 'material-ui/Button';

/**
 * Props for InstrumentCard
 * @interface InstrumentCardProps
 */
export interface InstrumentCardProps {
  id?: string;
  image: string;
  instrument: string;
  instructors: number;
}

/** 
 * InstrumentCard component
 */
const InstrumentCard: React.StatelessComponent<InstrumentCardProps> = props =>  {
    return (
      <div className="card nabi-display-inline-block">
        <Card>
        <CardMedia className="CardMedia box-shadow-none">
          <img src={require(`../../../../assets/images/${props.image}`)} alt=""/>
        </CardMedia>
        <CardContent>
        <Typography  variant="headline" component="h2">
         Learn {props.instrument}
         </Typography>
        <Typography component="p">
        {props.instructors} {props.instrument} instructor in your area.
        </Typography>
        <Button size="small" color="primary">
          GET STARED
        </Button>
        </CardContent>
        </Card>
      </div>
    );
};

export default InstrumentCard ;