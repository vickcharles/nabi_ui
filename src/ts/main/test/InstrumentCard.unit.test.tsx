import * as React from 'react';
import InstrumentCard from '../components/Homepage/InstrumentCard';
import { shallow } from 'enzyme';

describe('PopularInstruments', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(
      <InstrumentCard
        id=""
        index={0}
        image=""
        instrument=""
        instructors={0}
      />
    );
  });

  it('Renders the component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
