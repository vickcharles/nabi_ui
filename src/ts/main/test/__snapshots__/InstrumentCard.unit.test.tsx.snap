// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularInstruments Renders the component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(InstrumentCard)
    id=""
    image=""
    index={0}
    instructors={0}
    instrument=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "media": "InstrumentCard-media-1",
      },
      "id": "",
      "image": "",
      "index": 0,
      "instructors": 0,
      "instrument": "",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "media": "InstrumentCard-media-1",
        },
        "id": "",
        "image": "",
        "index": 0,
        "instructors": 0,
        "instrument": "",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
